R/zchunk_LA122.gasproc_refining.R

```diff
@@ -42,100 +42,370 @@ module_energy_LA122.gasproc_refining_DISABLED <- function(command, ...) {
+      filter(sector == "net_oil refining") %>%
+      filter(fuel == "refined liquids") %>%
+      left_join_error_no_match(select(filter(L1011.en_bal_EJ_R_Si_Fi_Yh, sector == "TPES", fuel == "refined liquids"), -sector), by = c("GCAM_region_ID", "fuel", "year")) %>%
+      select(-value.x) %>%
+      rename(value = value.y) %>%
+      bind_rows(filter(L1011.en_bal_EJ_R_Si_Fi_Yh,sector == "net_oil refining", fuel!= "refined liquids")) %>%
+      mutate(sector = "oil refining") %>%
+      mutate(fuel = if_else(fuel == "refined liquids", "oil", fuel)) -> L122.in_EJ_R_oilrefining_F_Yh
+
+    # Calculating region- and fuel-specific coefficients of crude oil refining
+    L122.in_EJ_R_oilrefining_F_Yh %>%
+     left_join(select(L122.out_EJ_R_oilrefining_Yh, -fuel), by = c("GCAM_region_ID", "sector", "year")) %>%
+      mutate(value = value.x/value.y) %>%
+      select(-value.x, -value.y) -> L122.IO_R_oilrefining_F_Yh
+
+    # COMBINE (RBIND) ALL CALIBRATED REFINERY INPUT AND OUTPUT TABLES
+    bind_rows( L122.out_EJ_R_oilrefining_Yh, L122.out_EJ_R_gtlctl_Yh, L122.out_EJ_R_biofuel_Yh ) -> L122.out_EJ_R_refining_F_Yh
+    bind_rows( L122.in_EJ_R_oilrefining_F_Yh, L122.in_EJ_R_gtlctl_F_Yh, L122.in_EJ_R_biofuel_F_Yh ) -> L122.in_EJ_R_refining_F_Yh
+
+
+    # Extra final step: calculate and create the derived crop inputs to the various first-generation biofuel technologies, for the AGLU processing
+    L122.in_EJ_R_biofuel_F_Yh %>%
+      left_join(select(calibrated_techs, sector, fuel, passthrough.sector = minicam.energy.input), by = c("sector", "fuel")) %>%
+      filter(passthrough.sector %in% A21.globaltech_coef$supplysector) -> L122.in_EJ_R_1stgenbio_F_Yh
+
+    A21.globaltech_coef %>%
+      rename(passthrough.sector = supplysector) %>%
+      select(passthrough.sector, minicam.energy.input) %>%
+      distinct(passthrough.sector, .keep_all = TRUE) %>%
+      right_join(L122.in_EJ_R_1stgenbio_F_Yh, by = "passthrough.sector") %>%
+      rename(GCAM_commodity = minicam.energy.input) -> L122.in_EJ_R_1stgenbio_F_Yh
+
+
+    # crop inputs to biodiesel are region-specific
+    L122.in_EJ_R_1stgenbio_F_Yh %>%
+      filter(passthrough.sector == "regional biomassOil") %>%
+      left_join(select(A_regions, GCAM_region_ID, biomassOil_tech), by = "GCAM_region_ID") %>%
+      select(-GCAM_commodity) %>%
+      rename(GCAM_commodity = biomassOil_tech) %>%
+      bind_rows(filter(L122.in_EJ_R_1stgenbio_F_Yh, passthrough.sector != "regional biomassOil")) %>%
+      filter(GCAM_commodity %in% FAO_ag_items_PRODSTAT$GCAM_commodity) %>%
+      select(GCAM_region_ID, sector, fuel, passthrough.sector, GCAM_commodity, year, value) -> L122.in_EJ_R_1stgenbio_F_Yh
+
+    # Interpolate coefs (using repeat_add_columns) to all historical periods, and then multiply by the input quantities
+    A21.globaltech_coef %>%
+      gather(Xyear, value, -supplysector, -subsector, -technology, -minicam.energy.input) %>%
+      filter(Xyear == "1971") %>%
```

Comment:
Well, but HISTORICAL_YEARS might change! So don't use the constant 1971, but rather calculate it as min(HISTORICAL_YEARS) please.